mlflow/server/js/src/experiment-tracking/components/modals/GenericInputModal.tsx

```diff
@@ -40,16 +40,13 @@ export class GenericInputModal extends Component<Props, State> {
     try {
       const values = await (this as any).formRef.current.validateFields();
 
-      // call handleSubmit from parent component, pass form values
-      // handleSubmit is expected to return a promise
-      return await this.props
+      await this.props
         .handleSubmit(values)
         .then(this.resetAndClearModalForm)
         .catch(this.handleSubmitFailure)
         .finally(this.onRequestCloseHandler);
     } catch (e) {
       this.setState({ isSubmitting: false });
-      return Promise.reject(e);
```

Comment:
As far as I know, no need to explicitly return promise. 
The original issue is caused by `return Promise.reject(e);` because the error was not handled by the caller. Also, returned value was not used.